corehq/apps/userreports/reports/data_source.py

```diff
@@ -123,22 +122,38 @@ def get_data(self, slugs=None):
             if self._order_by:
                 for col in reversed(self._order_by):
                     is_descending = col[1] == DESCENDING
-                    is_date = any(
-                        configured_indicator['datatype'] == 'date'
-                        for configured_indicator in self.config.configured_indicators
-                        if configured_indicator['column_id'] == col[0]
-                    )
-                    default_sort_by_date = (
-                        date(datetime.MINYEAR, 1, 1)
-                        if is_descending else date(datetime.MAXYEAR, 12, 31)
-                    )
-                    value = lambda x: x.get(col[0], None)
-                    sort_by_value = lambda x: (
-                        value(x)
-                        or (default_sort_by_date if is_date else value(x))
-                    )
+
+                    def sort_by(row):
+                        value = row.get(col[0], None)
+                        if not value:
+                            matching_indicators = filter(
+                                lambda configured_indicator: configured_indicator['column_id'] == col[0],
+                                self.config.configured_indicators
+                            )
+                            if not len(matching_indicators) == 1:
+                                raise UserReportsError(
```

Comment:
if we don't think we should ever fail loudly just because we can't sort (which seems right to me) then we should also trap this error outside the sort code